src/components/AuditLogsTable/AuditLogsTable.tsx

```diff
@@ -68,20 +89,30 @@ const AuditLogsTable = ({ showModel = false }: Props) => {
     (column) => showModel || column.accessor !== "model"
   );
 
-  const [, setQueryParams] = useQueryParams<{ page: string | null }>({
+  const [limit, setLimit] = useState<number>(DEFAULT_LIMIT_VALUE);
+  const [{ page }, setQueryParams] = useQueryParams<{ page: string | null }>({
     page: null,
   });
 
-  const fetchAuditEvents = useCallback(() => {
-    if (!wsControllerURL || !hasControllerConnection) {
-      return;
-    }
-    dispatch(jujuActions.fetchAuditEvents({ wsControllerURL }));
-  }, [dispatch, hasControllerConnection, wsControllerURL]);
+  const fetchAuditEvents = useCallback(
+    (limit: number) => {
+      if (!wsControllerURL || !hasControllerConnection) {
+        return;
+      }
+      dispatch(
+        jujuActions.fetchAuditEvents({
+          wsControllerURL,
+          limit: limit + 1,
+          offset: page === null ? 0 : (Number(page) - 1) * limit,
```

Comment:
If the default is set by the `useQueryParams` hook above then this can just be `offset: (Number(page) - 1) * limit`.